components/clp-package-utils/clp_package_utils/scripts/native/search.py

```
@@ -162,33 +171,82 @@ async def do_search(
     begin_timestamp: int | None,
     end_timestamp: int | None,
     ignore_case: bool,
-    max_num_results: int,
     path_filter: str | None,
     do_count_aggregation: bool | None,
     count_by_time_bucket_size: int | None,
 ):
-    db_monitor_task = asyncio.ensure_future(
-        run_function_in_process(
-            create_and_monitor_job_in_db,
-            db_config,
-            results_cache,
-            wildcard_query,
-            tags,
-            begin_timestamp,
-            end_timestamp,
-            ignore_case,
-            max_num_results,
-            path_filter,
-            do_count_aggregation,
-            count_by_time_bucket_size,
+    if do_count_aggregation is None:
+        host = None
+        for ip in set(socket.gethostbyname_ex(socket.gethostname())[2]):
+            host = ip
+            break
+        if host is None:
+            logger.error("Could not determine IP of local machine.")
+            return -1
+        try:
+            server = await asyncio.start_server(
+                client_connected_cb=worker_connection_handler,
+                host=host,
+                port=0,
+                family=socket.AF_INET,
+            )
+        except asyncio.CancelledError:
+            # Search cancelled
+            return
```

Any reason not to let the exception propagate to `asyncio.run`?